app/src/main/res/layout/activity_main.xml

```diff
@@ -16,14 +16,17 @@
     You should have received a copy of the GNU General Public License
     along with this program. If not, see <http://www.gnu.org/licenses/>.
 -->
-<android.support.v4.widget.DrawerLayout xmlns:android="http://schemas.android.com/apk/res/android"
+<android.support.v4.widget.DrawerLayout
+    xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:app="http://schemas.android.com/apk/res-auto"
     xmlns:tools="http://schemas.android.com/tools"
     android:id="@+id/drawer_layout"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
     android:fitsSystemWindows="true"
-    tools:openDrawer="start">
+    tools:openDrawer="start"
+    android:layout_gravity="right"
```

Comment:
again .. hardcoded right... should be "start"

Reply:
when we change language to Persian everything is fine except one and thats navigation drawer its better to open from rtl but its not i just want to change it from ltr to rtl

Reply:
For it to work properly and not break existing languages you should use the direction independent values "start" and "end" they are basically "left" and "right" but they switch around whenever an rtl language is active..